core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/JDBCUserStoreManager.java

```diff
@@ -1370,7 +1370,7 @@ public boolean doAuthenticate(String userName, Object credential) throws UserSto
                 Timestamp changedTime = rs.getTimestamp(6);
 
                 GregorianCalendar gc = new GregorianCalendar();
-                gc.add(GregorianCalendar.HOUR, -24);
+                gc.add(GregorianCalendar.HOUR, - getDefaultPasswordValidityPeriodInHours());
```

Comment:
since the validity period value is not subject to change during the runtime, we can use a static variable and reuse here instead of calling the method.

Reply:
You can define and call a method to resolve the value for the static variable during the class loading time

Reply:
changes made